### PR TITLE
RavenDB-7866

### PR DESCRIPTION
**Pull request by aviv86**

### CODE REVIEW

File: src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs [hunk 1]
```diff
@@ -136,10 +138,36 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
                     {
                         context.Visitor.Visit(node.Expression);
                     }
+
                     javascriptWriter.Write(".");
-                    javascriptWriter.Write(node.Member.Name == "Count" ? "length" : node.Member.Name);
+
+                    if (node.Member.Name == "Count" && IsCollection(node.Member.DeclaringType))
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs:144]: `Legnth` for array is already supposed, right?

File: src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs [hunk 2]
```diff
@@ -136,10 +138,36 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
...
+                        javascriptWriter.Write("length");
+                    }
+                    else
+                    {
+                        javascriptWriter.Write(node.Member.Name);
+                    }
                 }
             }
+
+            private static bool IsCollection(Type type)
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs:155]: You can replace all of this with: a check on `IEnumerable`

File: src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs (hunk 1 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs:144]: yes, Array.Length is supported

File: src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs (hunk 2 above)
Comment (aviv86) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs:155]: fixed

**Pull request merged**